source/ceylon/time/iso8601/parseTimeZone.ceylon

```diff
@@ -1,200 +1,67 @@
-import ceylon.time.base { ms = milliseconds }
 import ceylon.time.timezone {
-    timeZone,
-    TimeZone
+    TimeZone,
+    OffsetTimeZone,
+    timeZone
+}
+import ceylon.time.base {
+    milliseconds
 }
-
-
 
 "Timezone offset parser based on ISO-8601, currently it accepts the following 
  time zone offset patterns:
  
- - &plusmn;`[hh]:[mm]`, 
+ - &plusmn;`[hh]:[mm]`,
  - &plusmn;`[hh][mm]`, and 
  - &plusmn;`[hh]`.
  
  In addition, the special code `Z` is recognized as a shorthand for `+00:00`"
-shared TimeZone|ParserError parseTimeZone( String offset ) {
-    variable TzState state = TzInitial();
-    for( character in offset ) {
-        state = state.next( Chunk( character ) );
+shared TimeZone? parseTimeZone( String offset ) {
+    if(offset.equalsIgnoringCase("Z")) {
+        return timeZone.utc;
     }
-    state = state.next( eof );
-    if ( is TzFinal final = state ) {
-        value result = final.evaluate();
-        if ( result == 0 && offset.contains('-') ) {
-            return ParserError("Pattern not allowed by ISO-8601: '``offset``'!");
+    
+    value signal = offset.startsWith("+")
+            then 1 
+            else (offset.startsWith("-") 
+                then -1
+                else null);
+
+    value offsetWithoutSignal = offset.spanFrom(1);
+    Integer? hours;
```

Comment:
I don't think you need to declare these in advance here. This makes the code much longer and (at least to me) harder to follow.

Couldn't you just follow similar pattern as in `parseDate` and friends...?